File: cs50x/pset6/readability/readability.py
from cs50 import get_string


def main():
    # User Input
    text = get_string("Text: ")
    
    # Vairble to store letters
    nof_letters = 0
    # Vairble to store words
    nof_words = 1
    # Vairble to store sentences
    nof_sentences = 0
    
    # Determining the number of words, letters, and sentences
    for c in text:
        if (c.isalpha()):
            nof_letters += 1
        elif (c.isspace()):
            nof_words += 1
        elif ((c == ".") or (c == "!") or (c == "?")):
            nof_sentences += 1

    # Claculating L
    L = (nof_letters * 100) / nof_words
    # Calculating S
    S = (nof_sentences * 100) / nof_words
    # Getting the index
    Liau_Index = 0.0588 * L - 0.296 * S - 15.8
    
    # determining what the Liau Index is and giving the correct responses
    if (round(Liau_Index) < 1):
        print("Before Grade 1")
    elif ((round(Liau_Index) >= 1) and (round(Liau_Index) < 16)):
        print(f"Grade {round(Liau_Index)}")
    else:
        print("Grade 16+")

  
main()

File: cs50x/pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

int count_letters(string sentence);
int count_words(string sentence);
int count_sentences(string sentence);



int main(void)
{
    string text = get_string("Text: ");

    float L = ((float) count_letters(text) * 100) / (float) count_words(text);
    float S = ((float) count_sentences(text) * 100) / (float) count_words(text);
    float index = (0.0588 * L) - (0.296 * S) - 15.8;

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if ((1 < index) && (index < 16))
    {
        printf("Grade %.0f\n", index);
    }
    else
    {
        printf("Grade 16+\n");
    }
}



int count_letters(string sentence)
{
    int text_length = strlen(sentence);
    int number_of_letters = 0;
    for (int i = 0; i <= text_length; i++)
    {
        if (isalpha(sentence[i]))
        {
            number_of_letters++;
        }
    }
    return number_of_letters;
}



int count_words(string sentence)
{
    int text_length = strlen(sentence);
    int number_of_words = 0;
    for (int i = 0; i <= text_length; i++)
    {
        if (isspace(sentence[i]))
        {
            number_of_words++;
        }
    }
    return number_of_words + 1;
}



int count_sentences(string sentence)
{
    int text_length = strlen(sentence);
    int number_of_sentences = 0;
    for (int i = 0; i <= text_length; i++)
    {
        if ((int) sentence[i] == 33 || (int) sentence[i] == 46 || (int) sentence[i] == 63)
        {
            number_of_sentences++;
        }
    }
    return number_of_sentences;
}



